### DISCUSSION TITLE
Capping Memory Usage

### DISCUSSION THREAD

**Discussion by victorstewart** [hse-project/hse#26] [Q&A] [answered]
is there any way to put a ceiling on the memory usage of HSE without using some operating system mechanism to limit the memory resources of the process as a whole?

The latter I can and will do, but I was hoping to section off a part of that memory exclusively for my application running HSE inside it.

If not, does its cache size just grow without bound until it has consumed the available memory on the system as you `hse_kvs_get()` more and more data into memory?

**Comment by smoyerx** [comment, marked as answer]
There is not currently an explicit method to cap the user-space memory consumed by a KVDB.  Much of the total memory consumed by KVDB is in the page cache, never dirty, and the kernel can always steal it as needed.

We haven't observed memory consumption to be an issue, but I understand that some use-cases may want the ability to cap it.

**Comment by victorstewart** [comment]
okay thanks!
